fe/fe-core/src/main/java/com/starrocks/connector/hive/HiveConnectorScanRangeSource.java

```diff
@@ -0,0 +1,409 @@
+                                                      long offset, long length) {
+            PartitionAttachment attachment = (PartitionAttachment) remoteFileInfo.getAttachment();
+            TScanRangeLocations scanRangeLocations = new TScanRangeLocations();
+
+            THdfsScanRange hdfsScanRange = new THdfsScanRange();
+            hdfsScanRange.setRelative_path(fileDesc.getFileName());
+            hdfsScanRange.setOffset(offset);
+            hdfsScanRange.setLength(length);
+            hdfsScanRange.setPartition_id(attachment.partitionId);
+            hdfsScanRange.setFile_length(fileDesc.getLength());
+            hdfsScanRange.setModification_time(fileDesc.getModificationTime());
+
+            RemoteFileInputFormat fileFormat = remoteFileInfo.getFormat();
+            hdfsScanRange.setFile_format(fileFormat.toThrift());
+            if (fileFormat.isTextFormat()) {
+                hdfsScanRange.setText_file_desc(fileDesc.getTextFileFormatDesc().toThrift());
+            }
+
+            if (attachment.dataCacheOptions != null) {
+                TDataCacheOptions tDataCacheOptions = new TDataCacheOptions();
+                tDataCacheOptions.setPriority(attachment.dataCacheOptions.getPriority());
+                hdfsScanRange.setDatacache_options(tDataCacheOptions);
+            }
+
+            TScanRange scanRange = new TScanRange();
+            scanRange.setHdfs_scan_range(hdfsScanRange);
+            scanRangeLocations.setScan_range(scanRange);
+
+            if (blockDesc.isPresent()) {
+                if (blockDesc.get().getReplicaHostIds().length == 0) {
+                    String message = String.format("hdfs file block has no host. file = %s/%s",
+                            remoteFileInfo.getFullPath(), fileDesc.getFileName());
+                    throw new StarRocksPlannerException(message, ErrorType.INTERNAL_ERROR);
+                }
+
+                for (long hostId : blockDesc.get().getReplicaHostIds()) {
+                    String host = blockDesc.get().getDataNodeIp(hostId);
+                    TScanRangeLocation scanRangeLocation = new TScanRangeLocation(new TNetworkAddress(host, -1));
+                    scanRangeLocations.addToLocations(scanRangeLocation);
+                }
+            } else {
+                TScanRangeLocation scanRangeLocation = new TScanRangeLocation(new TNetworkAddress("-1", -1));
+                scanRangeLocations.addToLocations(scanRangeLocation);
+            }
+            buffer.add(scanRangeLocations);
+        }
+
+    }
+
+    public Iterator<TScanRangeLocations> createScanRangeIterator(RemoteFileInfo remoteFileInfo) {
+        return new ScanRangeIterator(remoteFileInfo);
+    }
+
+    private void updateBackendSplitFile(RemoteFileInfo partition) {
+        ConnectContext connectContext = ConnectContext.get();
+        if (connectContext == null) {
```

Comment:
connectcontext will be null in some background thread like refresing metadata cache.